docs/devices_api.yml

```diff
@@ -1,99 +1,121 @@
+                                 eyJleHAiOjE0NzYxMTkxMzYsImp0aSI6Ijg1NGIzMTA5LTQ4NjItNGEyNS1h
+                                 MWZiLWYxMTE2MWNlN2E4NCIsImlzcyI6Ik1lbmRlciIsInN1YiI6IjlmNzM2
+                                 YmNiMjhiZmFhOTg5YjVmNWUxNDA5ZGJmMGVhYzdhNjYxMjZiNjMyZDAzYWYwZ
+                                 mUzNGFjMjhiZjRhNzIifQ.
+                                 PArg_WuoQkOiJ4kDoHYbQRjnxykeF1lIlsgJfUryhivnip2AHz5bkxxaxF20X
+                                 Tq9mIzSDonTSukfOtkaxJTZXjCMHjgh50iwa6_pUivIYWsIJW2O9t_M9T_SC-
+                                 7Xu7IhE_iKQFb2NXxVfAG4nZKrheUM4MJBt8SxCawT2EOPopiLeIC6MOFBu_s
+                                 Pa9RsagKSZCRaLTBWVhmEGbfn19tLOX3Z06DZql61G-VY-YuyOlBjpEsCc4Hi
+                                 A1cXIdncCZKugrONOa44_m4yx0VsgRg4jCd2VO-Is-A96Jw3zkZshoD2cPXVS
+                                 KAhFdhHja447ftuYYRq9kIQghKi3hfsPgyFZQ
         401:
-          description: The device cannot be granted authentication (e.g. not accepted
-                       by the user yet, or explicitly rejected).
+          description: |
+                The device cannot be granted authentication. There are multiple possible reasons, e.g.:
+                * the device hasn't been accepted yet
+                * the device has been explicitly rejected
+                * key/signature don't match
+
+                See the error message for details.
+          schema:
+            $ref: '#/definitions/Error'
         400:
-          description: Missing/malformed request params or body.
+          description: Missing or malformed request params or body. See the error message for details.
           schema:
             $ref: '#/definitions/Error'
         500:
-          description: Unexpected error
+          description: Internal server error.
           schema:
             $ref: '#/definitions/Error'
 
 definitions:
-  AuthRequestNew:
+  AuthRequest:
     type: object
     properties:
       id_data:
         type: string
-        description: Vendor-specific JSON representation of device identity, encrypted
-                     with the tenant's public key.
-
-                     In reference implementation, it is a JSON structure
-                     with vendor-selected fields, such as MACs, serial numbers, etc.
-      tenant_token:
-        type: string
-        description: A tenant's unique token (issued by Mender via web GUI/API).
+        description: Vendor-specific JSON representation of the device identity data (MACs, serial numbers, etc.).
       pubkey:
         type: string
         description: The device's public key, generated by the device or pre-provisioned by the vendor.
+      seq_no:
+        type: integer
+        format: uint64
+        description: |
+            An ever-increasing sequence number, recorded by the server
+            upon each authentication request to prevent replay attacks.
+
+            The device may choose any non-negative number for the bootstrap request.
```

Comment:
well, it will quickly break then, that's the only corner case.

anyway, we don't handle it.

should we recommend a small number then? because I don't see a way of solving this, there's always be a possiblity of overflow